tests/test_flow/test_persistence_fuzz.py

```diff
@@ -270,21 +332,41 @@ def run(self, n_iterations):
                     self.model.expected_entity_value(affected_name)
                 )
 
-            if make_func_change or self._versioning_mode == 'auto':
-                expected_called_names = self.model.entity_names(
+            if make_func_change:
+                entity_names_in_between = set(self.model.entity_names(
                     downstream_of=updated_name,
                     upstream_of=affected_names,
-                )
-                assert (
-                    list(sorted(self.model.called_entity_names())) ==
-                    list(sorted(expected_called_names))
-                )
+                ))
+                nondeterministic_ancestors = set()
+                for entity_name in entity_names_in_between:
```

Comment:
Could we just use `nondeterministic_ancestors = self.model.nondeterministic_upstream_entity_names(affected_names)` here? If some entity X is upstream of any in-between entity, doesn't that imply that X is also upstream of some affected entity?

If so, I wonder if we could factor out the common parts of these two cases (`make_func_change` and `not make_func_change`).